app/models/rule.rb

```diff
@@ -113,14 +117,34 @@ def self.revert(rule, audit_id, fields, audit_comment)
 
   private
 
-  def error_if_locked
-    # locked = current update
-    # locked_was = before
-    # If the previous state was not locked, updates can be made.
-    return unless locked_was
+  def cannot_be_locked_and_under_review
+    return unless locked && review_requestor_id.present?
+
+    errors.add(:base, 'Control cannot be under review and locked at the same time.')
+  end
 
-    # If the previous state was locked, error
-    raise(RuleLockedError, id) if locked_was
+  ##
+  # Check to ensure that "review fields" are not changed
+  # in the same `.save` action as any "non-review fields"
+  def review_fields_cannot_change_with_other_fields
+    review_fields = Set.new(%w[review_requestor_id locked updated_at])
+    any_review_fields_changed = changed.any? { |field| review_fields.include? field }
+    any_non_review_fields_changed = changed.any? { |field| review_fields.exclude? field }
+    # Break early if review and non-review fields have not changed together
+    return unless any_review_fields_changed && any_non_review_fields_changed
+
+    raise(RuleReviewError, 'Cannot update review-related attributes with other non-review-related attributes')
```

Comment:
I don't understand why these are errors when they could just be errors that prevent save. By making them errors you're having to deal with 2 different codepaths in the controller, one for exceptions and one for validation errors.